### ISSUE TITLE
uploadFiles and downloadFile have different error handling logic. 

### ISSUE THREAD

**Issue by Yupeng-li** [birdofpreyru/react-native-fs#56]
Hi, 

I noticed that in `uploadFiles` the http status code that is greater than or equal to 400 is handled internally and is thrown as an error. See [code](https://github.com/birdofpreyru/react-native-fs/blob/v2.27.1/src/index.ts#L383)

But `downloadFile` doesn't do the same. See [code](https://github.com/birdofpreyru/react-native-fs/blob/v2.27.1/src/index.ts#L193). 

Also, the Error thrown by `uploadFiles` has an untyped field `result`. It's useful, but since it's not typed and is optional, we have to type it by ourselves. 

I think it's better that they are consistent and there is a dedicated type of Error for http error responses.

**Comment by birdofpreyru** [comment]
Yes, I agree. `uploadFiles()` got recently updated, because of #48 / PR #49, but I had not arrived to update `downloadFile()` in the similar way. Please, submit a PR if you need it and have capacity to do it.
